client.go

```diff
@@ -219,6 +221,40 @@ func (co *Conn) ReadMsg() (*Msg, error) {
 	return m, err
 }
 
+// ReadMsgBytes reads a message bytes from the connection. Parses and fills
+// dns message wire header (passing nil would skip header parsing) and
+// returns message bytes to process them later.
+//
+// Note that this function would not be able to report TSIG error or
+// check it got actual DNS payload.
+func (co *Conn) ReadMsgBytes(hdr *Header) ([]byte, error) {
```

Comment:
Sorry, I don't understand that remark. At most this function read 12 bytes from a message?